src/Provider/RequestHandler.php

```diff
@@ -78,10 +79,35 @@ public function dispatch( $wp ) {
 			$this->request->set_url_params( $wp->query_vars['satispress_params'] );
 		}
 
-		if ( $this->check_authentication() ) {
-			$this
-				->get_route_controller( $route )
-				->handle_request( $this->request );
+		try {
+			if ( $this->check_authentication() ) {
+				$this
+					->get_route_controller( $route )
+					->handle_request( $this->request );
+			}
+		} catch ( HTTPException $e ) {
+			$message = $e->getMessage();
+
+			// Show extra exception data to administrators or when in debugging mode.
+			if (
```

Comment:
Let's move this conditional into its own method (or standalone function if it's likely to be used elsewhere, so that we can remove the comment and make it more self-documenting:

```php
if ( $this->can_show_extra_exception_data() ) {
...
}
```

It's then only one thing to monkey patch / mock when writing tests to determine the message output.

Reply:
There is also currently 11 lines that build up that extra exception message, which is polluting the `dispatch()` method. Should be able to move this whole condition block elsewhere, like overwriting the `getMessage()` method for the HTTP exception.